src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs

```diff
@@ -344,22 +348,24 @@ public void Login()
         public bool OpenRepository()
         {
             var old = Repositories.FirstOrDefault(x => x.Equals(Holder.ActiveRepo));
-            // open the solution selection dialog when the user wants to switch to a different repo
-            // since there's no other way of changing the source control context in VS
             if (!Equals(SelectedRepository, old))
             {
-                if (ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1)))
+                var opened = vsServices.TryOpenRepository(SelectedRepository.LocalPath);
+                if (!opened)
                 {
-                    ServiceProvider.TryGetService<ITeamExplorer>()?.NavigateToPage(new Guid(TeamExplorerPageIds.Home), null);
-                    return true;
+                    // Fall back to making the user select a solution to open.
+                    opened = ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1));
                 }
-                else
+
+                if (!opened)
                 {
-                    SelectedRepository = old;
                     return false;
```

Comment:
@jcansdale `SelectedRepository` is set to whatever the user double-clicks on the list (binding here https://github.com/github/VisualStudio/blob/ae375bf569d129a768612919e86f7ddeb39964aa/src/GitHub.VisualStudio.UI/UI/Views/GitHubConnectContent.xaml#L161). If we fail to actually act on the new selection, then we need to reset it back to the current one (cached here https://github.com/github/VisualStudio/blob/ae375bf569d129a768612919e86f7ddeb39964aa/src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs#L350 from the currently active repository) so the UI doesn't look like we actually did open the thing when we didn't.